### ISSUE TITLE
Updater GUI: Clarify meaning of "MAYBE" status (including how it's different from "NO")

### ISSUE THREAD

**Issue by moonlitOrca** [QubesOS/qubes-issues#8511]
[How to file a helpful issue](https://www.qubes-os.org/doc/issue-tracking/)

### Qubes OS release
Qubes OS 4.2 RC3

### Brief summary
The New Qubes Update tool is truly a stellar experience and meaningful upgrade over the last version.  In particular, the new system of stating "YES", "MAYBE", and "NO" under the updates available column is a very nice touch.  However, it seems that the only meaningful tag is "YES" as this is correctly changed once updates are detected for a Qube.  I assumed "MAYBE" would be the default tag if updates were not yet checked very recently, but it seems that sometimes "MAYBE" is used and sometimes "NO" is used regardless of whether updates have been checked for or not for a long time (I'm uncertain of the timeframe).

### Steps to reproduce
Use Qubes Update regularly and see that sometimes qubes that do actually have updates pending are marked "NO" updates available.  And sometimes they are marked "MAYBE", even if the Qubes Update Tool has not checked for many days.

### Expected behavior

If Qubes OS has not checked for updates yet within a reasonably short time span (perhaps 1 or 2 days), all VMs should read "MAYBE" to convey to the user that state of uncertainty.  If updates have been checked, that should switch to either "YES" if updates are found (this seems to work) or "NO" if updates are not found.  This would reflect what a user would likely intuit and expect regarding the meaning of "MAYBE" and "NO".

### Actual behavior
The various qubes will correctly read "YES" if updates are found, but the other qubes which have not been checked, will sometimes say "MAYBE" and sometimes say "NO" without reflecting the actual status of whether or not updates have been checked in the past few days.

**Comment by andrewdavidwong** [comment]
Sounds somewhat related to #7818.

> Use Qubes Update regularly and see that sometimes qubes that do actually have updates pending are marked "NO" updates available.

Alternative idea: Somehow communicate to the user that the "NO" actually means something like, "NO as of the last update check for this template, the date of which you can see in the column to the right."

In other words, the last time we checked for updates for this template, none were found, so we're putting a "NO" here. But, of course, updates can become available at any time, so the user should understand that the "NO" label always means "NO as of the last time we checked."

> If Qubes OS has not checked for updates yet within a reasonably short time span (perhaps 1 or 2 days), all VMs should read "MAYBE" to convey to the user that state of uncertainty.

This doesn't seem like an ideal solution, because updates can become available one hour or even one second after the last check, so "1 or 2 days" is quite arbitrary.

**Comment by moonlitOrca** [comment]
> Sounds somewhat related to #7818.
> 
> > Use Qubes Update regularly and see that sometimes qubes that do actually have updates pending are marked "NO" updates available.
> 
> So, the UX challenge is to somehow communicate to the user that the "NO" does _not_ mean, "There are no updates available right now" (which would be false). Instead, it actually means something like, "There were no updates available the last time we checked, which was on the date in the column to the right" (which is true).
> 
> > If Qubes OS has not checked for updates yet within a reasonably short time span (perhaps 1 or 2 days), all VMs should read "MAYBE" to convey to the user that state of uncertainty.
> 
> This doesn't seem like an ideal solution, because updates can become available one hour or even one second after the last check, so "1 or 2 days" is quite arbitrary.

Agreed.  It is arbitrary, mostly taken from past experience with that being the way Windows Update handles the timeframe for saying "no updates", and that seems to make sense to many users.  That being said, unless the updater checks each time it is opened to update the status, any timeframe seems arbitrary, as you say.

At least to my intuition as a user, the "NO" updates available directly next to "last time checked" is reasonable (as the interface currently has), except that the "MAYBE" exists as a flag option.  What is the "MAYBE" option designed to indicate, if not that updates have not been checked quite recently?  Because that's what "threw me off" as assuming the "NO" had been checked very recently.

Perhaps one other option to consider: what if there was a button to manually "Check for Updates"?  This is a common option in may operating systems, and would allow Qubes OS users to have an accurate answer to the question, but also not automatically apply the update.  At this time, the only graphical way to check for updates is actually by performing the update, right?  I image running the VM based on a certain template might be a workaround for that (forcing Qubes OS to check in the background for that VM), but that is not exactly an intuitive solution.

**Comment by andrewdavidwong** [comment]
> At least to my intuition as a user, the "NO" updates available directly next to "last time checked" is reasonable (as the interface currently has), except that the "MAYBE" exists as a flag option. What is the "MAYBE" option designed to indicate, if not that updates have not been checked quite recently? Because that's what "threw me off" as assuming the "NO" had been checked very recently.

These two points seem to contradict one another:

1. "NO" is already next to "last time checked," so you know the specific time to which the "NO" is indexed.
2. You assumed that the "NO" is the result of a very recent check.

**Comment by moonlitOrca** [comment]
I will try to say it a different way: The confusion here results not just about the "NO" flag, but also about the "MAYBE" flag. In fact, to me, the "MAYBE" status is what causes the confusion primarily. 

From my intuitive sense as a new user of the new "Qubes Updater" tool, I saw that there are three possible states for the updates check to be in.  the "YES" state is obvious; it must have been checked and updates were discovered (as of last check).  The " MAYBE" state is ambiguous; I assumed that meant that updates had *not* been checked recently and that the user should check those qubes for updates.  What else could it mean?

I normally would assume that the "NO" option means there were no updates as of last check date.  This is the intention of the GUI, I believe, and this is reasonable.  *But*, if there is also a "MAYBE" status that is possible, then the "NO" status should mean something more concrete than the "MAYBE" status indicates.  So, if "MAYBE" means that updates are not certain because there has not been a recent check, then "NO" must mean something else.  It must mean that there is somehow more certainty that there are no updates available.  

Does that make sense?  In my mind, the "MAYBE" status needs better explanation, because it causes me to feel that the "NO" status means something *more* concrete.

**Comment by marmarek** [comment]
> The " MAYBE" state is ambiguous; I assumed that meant that updates had _not_ been checked recently and that the user should check those qubes for updates.

Yes, that's exactly what it means. The meaning of "recently" is configurable in the updater itself (the gear icon in the right top corner), and AFAIR the default is 7 days. It's meant to handle templates that are not used often, or where all qubes based on it as offline (so, cannot check updates for this template).

**Comment by moonlitOrca** [comment]
> It's meant to handle templates that are not used often, or where all qubes based on it as offline (so, cannot check updates for this template).

This seems a very reasonable feature to have implemented then in those use-cases.  Thank you for taking the time to clarify for me.  :-).  

I think @andrewdavidwong just updated the title to a logical course of action then: we probably just need to clarify for the user what the "MAYBE" means somewhere in the GUI since it is a fairly unusual term to be used in an update manager and it doesn't seem to be explained anywhere (leading to my confusion).  Or perhaps use a different term for that situation of "Not Checked Recently".  Not sure what would be quite as pithy as "MAYBE"....  I will give it some thought.

Thanks for the help both of you!  I am happy to test or provide any ideas if it would be helpful.

**Comment by ben-grande** [comment]
> The " MAYBE" state is ambiguous; I assumed that meant that updates had _not_ been checked recently and that the user should check those qubes for updates. What else could it mean?

That is correct.

A duplicate of this issue was closed with linked PR, therefore I am closing this as duplicate.
